src/server/terrain/structure_building_blocks.zig

```diff
@@ -0,0 +1,149 @@
+const std = @import("std");
+
+const main = @import("root");
+const ZonElement = main.ZonElement;
+const Blueprint = main.blueprint.Blueprint;
+const List = main.List;
+const ListUnmanaged = main.ListUnmanaged;
+const AliasTable = main.utils.AliasTable;
+const Neighbor = main.chunk.Neighbor;
+const Block = main.blocks.Block;
+const Degrees = main.rotation.Degrees;
+const NeverFailingAllocator = main.heap.NeverFailingAllocator;
+
+var arena = main.heap.NeverFailingArenaAllocator.init(main.globalAllocator);
+const arenaAllocator = arena.allocator();
+
+var blueprintCache: std.StringHashMapUnmanaged(*[4]BlueprintEntry) = .{};
+var childrenToResolve: List(struct {parentId: []const u8, colorName: []const u8, colorIndex: usize, childIndex: usize, structureId: []const u8}) = undefined;
+
+const originBlockStringId = "cubyz:sbb/origin";
+var originBlockNumericId: u16 = 0;
+
+// Maps global child block numeric ID to index used to locally represent that child block.
+var childBlockNumericIdMap: std.AutoHashMapUnmanaged(u16, u16) = .{};
+var childBlockStringId: ListUnmanaged([]const u8) = .{};
+
+const BlueprintEntry = struct {
+	blueprint: Blueprint,
+	originBlock: StructureBlock,
+	childBlocks: []StructureBlock,
+
+	const StructureBlock = struct {
+		x: i32,
+		y: i32,
+		z: i32,
+		index: u32,
```

Comment:
This can also be `u16`, since we cannot have 65536 sbbChild blocks anyways.

Reply:
Depends if we restrict x, y, z to `u16`, if we don't it will be 4 bytes aligned anyway.